objcgen/CachedValue.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace Embeddinator
+{
+	class CachedValue<T> where T : class
+	{
+		Func<T> CalculateProc;
+		T LastCalculatedValue;
+
+		public bool IsFrozen { get; private set; }
+		public void Freeze () => IsFrozen = true;
+
+		public CachedValue (Func<T> calculateProc)
+		{
+			CalculateProc = calculateProc;
+		}
+
+		public T Value
+		{
+			get
+			{
+				if (IsFrozen) {
+					if (LastCalculatedValue == null)
+						LastCalculatedValue = CalculateProc ();
+					return LastCalculatedValue;
+				}
+				else {
+					return CalculateProc ();
+				}
+			}
+		}
+	}
+}
```

Comment:
Turns out VSfM and XS don't share settings, and VSfM was configured all sorts of wrong. No wonder I've screwed this up a bunch recently....